resources/qml/Cura.qml

```diff
@@ -369,16 +387,59 @@ UM.MainWindow
             {
                 id: sidebar
 
+                property bool collapsed: false;
+                property var initialWidth: UM.Theme.getSize("sidebar").width;
+
+                function callExpandOrCollapse(){
+                    if(collapsed){
+                        sidebar.visible = true
+                        sidebar.initialWidth = UM.Theme.getSize("sidebar").width
+                        viewportRect = Qt.rect(0, 0, (base.width - sidebar.width) / base.width, 1.0)
+                        expandSidebarAnimation.start();
+                    }else{
+                        viewportRect = Qt.rect(0, 0, 1, 1.0)
+                        collapsSidebarAnimation.start();
+                    }
+                    collapsed = !collapsed;
+                    UM.Preferences.setValue("general/sidebar_collaps", collapsed);
+                }
+
                 anchors
                 {
-                    top: topbar.bottom
+                    top: topbar.top
                     bottom: parent.bottom
-                    right: parent.right
                 }
 
-                width: UM.Theme.getSize("sidebar").width
-
+                width: initialWidth
+                x: base.width - sidebar.width
                 source: UM.Controller.activeStage.sidebarComponent
+
+                NumberAnimation {
+                    id: collapsSidebarAnimation
+                    target: sidebar
+                    properties: "x"
+                    to: base.width
+                    duration: 500
+                }
+
+                NumberAnimation {
+                    id: expandSidebarAnimation
+                    target: sidebar
+                    properties: "x"
+                    to: base.width - sidebar.width
+                    duration: 500
```

Comment:
Let's make the duration either very fast (100ms) or remove the animation all together.